src/client/api/mgmt.c

```diff
@@ -112,24 +124,82 @@ daos_pool_create(uint32_t mode, uid_t uid, gid_t gid, const char *grp,
 		return -DER_INVAL;
 	}
 
+	entries = (pool_prop == NULL) ? 0 : pool_prop->dpp_nr;
+
+	/*
+	 * TODO: remove uid/gid input params and use euid/egid instead
+	 */
+	if (!daos_prop_has_entry(pool_prop, DAOS_PROP_PO_OWNER)) {
+		rc = daos_acl_uid_to_principal(uid, &owner);
+		if (rc) {
+			D_ERROR("Invalid uid\n");
+			return rc;
+		}
+
+		entries++;
+	}
+
+	if (!daos_prop_has_entry(pool_prop, DAOS_PROP_PO_OWNER_GROUP)) {
+		rc = daos_acl_gid_to_principal(gid, &owner_grp);
+		if (rc) {
+			D_ERROR("Invalid gid\n");
+			D_GOTO(out, rc);
+		}
+
+		entries++;
+	}
+
+	if (pool_prop == NULL || entries > pool_prop->dpp_nr) {
+		uint32_t idx = 0;
+
+		final_prop = daos_prop_alloc(entries);
+		if (pool_prop != NULL) {
+			rc = daos_prop_copy(final_prop, pool_prop);
+			if (rc)
+				D_GOTO(out, rc);
+			idx = pool_prop->dpp_nr;
+		}
+		if (owner != NULL) {
+			final_prop->dpp_entries[idx].dpe_type =
+				DAOS_PROP_PO_OWNER;
+			final_prop->dpp_entries[idx].dpe_str = owner;
+			owner = NULL;
+			idx++;
+		}
+
+		if (owner_grp != NULL) {
+			final_prop->dpp_entries[idx].dpe_type =
+				DAOS_PROP_PO_OWNER_GROUP;
+			final_prop->dpp_entries[idx].dpe_str = owner_grp;
+			owner_grp = NULL;
+			idx++;
+		}
+
+	} else {
+		final_prop = pool_prop;
+	}
+
 	rc = dc_task_create(dc_pool_create, NULL, ev, &task);
 	if (rc)
-		return rc;
+		D_GOTO(out, rc);
 
 	args = dc_task_get_args(task);
-	args->mode	= mode;
-	args->uid	= uid;
-	args->gid	= gid;
 	args->grp	= grp;
 	args->tgts	= tgts;
 	args->dev	= dev;
 	args->scm_size	= scm_size;
 	args->nvme_size	= nvme_size;
-	args->prop	= pool_prop;
+	args->prop	= final_prop;
 	args->svc	= svc;
 	args->uuid	= uuid;
 
-	return dc_task_schedule(task, true);
+	rc = dc_task_schedule(task, true);
+
+out:
+	daos_prop_free(final_prop);
```

Comment:
If the call to daos_pool_create is done asynchronously, that free call would make args->prop invalid and will cause segfaults in the dc_pool_create, no? I think you will need to free it in the completion callback of the pool create operation (pool_create_cp in src/mgmt/cli_pool.c)
I'm not sure about owner or owner_grp if they are used internally or not, in which case they need to be freed after the operation is done too.

Reply:
Fixed